src/main.rs

```diff
@@ -16,142 +18,149 @@ fn create_static_sampler(
-                2,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipLinear,
-                IRTextureAddressMode::IRTextureAddressModeClamp,
-                3,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterMinMagMipLinear,
-                IRTextureAddressMode::IRTextureAddressModeBorder,
-                4,
-                None,
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterAnisotropic,
-                IRTextureAddressMode::IRTextureAddressModeWrap,
-                5,
-                Some(2),
-            ),
-            create_static_sampler(
-                IRFilter::IRFilterAnisotropic,
-                IRTextureAddressMode::IRTextureAddressModeWrap,
-                6,
-                Some(4),
-            ),
-        ];
+        // Create an explicit root signature layout
+        let mut parameters = create_root_parameters();
+        let mut static_samplers = create_static_samplers();
 
         let desc_1_1 = IRRootSignatureDescriptor1 {
-            flags: IRRootSignatureFlags::IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed,
-            num_parameters: parameters.len() as u32,
-            p_parameters: parameters.as_ptr(),
-            num_static_samplers: static_samplers.len() as u32,
-            p_static_samplers: static_samplers.as_ptr(),
+            Flags: IRRootSignatureFlags::IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed,
+            NumParameters: parameters.len() as u32,
+            pParameters: parameters.as_mut_ptr(),
+            NumStaticSamplers: static_samplers.len() as u32,
+            pStaticSamplers: static_samplers.as_mut_ptr(),
         };
 
         let desc = IRVersionedRootSignatureDescriptor {
             version: IRRootSignatureVersion::IRRootSignatureVersion_1_1,
-            u: IRVersionedRootSignatureDescriptor_u { desc_1_1 },
+            u_1: IRVersionedRootSignatureDescriptor_u { desc_1_1 },
         };
 
-        let root_sig = IRRootSignature::create_from_descriptor(&lib, &desc)?;
+        let root_sig = IRRootSignature::create_from_descriptor(&compiler, &desc)?;
+        compiler.set_global_root_signature(&root_sig);
 
-        let egui_update =
-            include_bytes!("D:/breda/crates/breda-egui/assets/shaders/egui_update.cs.dxil");
-        // let memcpy = include_bytes!("C:/Users/Jasper/traverse/breda/apps/cs-memcpy/assets/shaders/memcpy.cs.dxil");
+        // Load DXIL
+        let dxil = include_bytes!("assets/memcpy.cs.dxil");
+        let obj = IRObject::create_from_dxil(&compiler, dxil)?;
 
-        let mut mtl_binary = IRMetalLibBinary::new(&lib)?;
+        // Convert to Metal
+        let mut mtl_binary = IRMetalLibBinary::new(&compiler)?;
+        let mtllib = compiler
+            .alloc_compile_and_link(CStr::from_bytes_with_nul_unchecked(b"main\0"), &obj)?;
```

Comment:
Since we don't have an MSRV here:
```suggestion
            .alloc_compile_and_link(c"main", &obj)?;
```